theblues/charmstore.py

```diff
@@ -198,16 +230,20 @@ def files(self, entity_id, manifest=None, filename=None, read_file=False):
         @param filename The name of the file in the archive to get.
         @param read_file Whether to get the url for the file or the file
         contents.
+        @param channel optional channel information.
         '''
         if manifest is None:
-            manifest_url = '%s/%s/meta/manifest' % (self.url,
-                                                    _get_path(entity_id))
+            manifest_url = '{}/{}/meta/manifest'.format(self.url,
```

Comment:
Again here, yeah we should have an `add_channel(url, channel=None) -> url` function somewhere.